sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java

```diff
@@ -567,13 +575,7 @@ private RntbdRequestRecord addPendingRequestRecord(final ChannelHandlerContext c
             final Timeout pendingRequestTimeout = record.newTimeout(timeout -> {
 
                 // We don't wish to complete on the timeout thread, but rather on a thread doled out by our executor
-                final EventExecutor executor = context.executor();
-
-                if (executor.inEventLoop()) {
-                    record.expire();
-                } else {
-                    executor.next().execute(record::expire);
-                }
+                requestExpirator.execute(record::expire);
```

Comment:
BacklogItem: gate the validation and detect future regressions

Reply:
ACK - but let's please track this in work items - not in this PR. We need to make progress on this PR quickly to ship the hotfix.